src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -350,53 +366,61 @@ class Distributor
     {
       if (_sources[i] != comm_rank)
       {
-        auto const message_size =
-            _src_counts[i] * num_packets * sizeof(ValueType);
-        auto const receive_buffer_ptr =
-            imports.data() + _src_offsets[i] * num_packets;
+        auto const receive_buffer_ptr = imports_comm.data() + _src_offsets[i];
+        auto const message_size = _src_counts[i] * sizeof(ValueType);
         requests.emplace_back();
         MPI_Irecv(receive_buffer_ptr, message_size, MPI_BYTE, _sources[i], 123,
                   _comm, &requests.back());
       }
     }
 
-    // make sure the data in dest_buffer has been copied before sending it.
-    if (permutation_necessary)
```

Comment:
Because it is for both GPU-aware and non-GPU aware paths. For non-GPU aware, we need to make sure the (parts of) export data are finished copying to the host. I think the right check for this would be "(if gpu-aware and permutation is necessary) or (not gpu-aware)".